autogen/agentchat/contrib/vectordb/pgvectordb.py

```diff
@@ -0,0 +1,710 @@
+        """
+        Delete the entire collection.
+
+        Args:
+            collection_name (Optional[str]): The name of the collection to delete.
+
+        Returns:
+            None
+        """
+        if collection_name:
+            self.name = collection_name
+        cursor = self.client.cursor()
+        cursor.execute(f"DROP TABLE IF EXISTS {self.name}")
+        cursor.close()
+
+    def create_collection(self, collection_name: str = None):
+        """
+        Create a new collection.
+
+        Args:
+            collection_name (Optional[str]): The name of the new collection.
+
+        Returns:
+            None
+        """
+        if collection_name:
+            self.name = collection_name
+        cursor = self.client.cursor()
+        cursor.execute(f"DROP TABLE IF EXISTS {self.name}")
+        cursor.execute(
+            f"CREATE TABLE {self.name} ("
+            f"documents text, id CHAR(8) PRIMARY KEY, metadatas JSONB, embedding vector(384));"
+            f"CREATE INDEX "
+            f'ON {self.name} USING hnsw (embedding vector_l2_ops) WITH (m = {self.metadata["hnsw:M"]}, '
+            f'ef_construction = {self.metadata["hnsw:construction_ef"]});'
+        )
+        cursor.close()
+
+
+class PGVectorDB(VectorDB):
+    """
+    A vector database that uses PGVector as the backend.
+    """
+
+    def __init__(
+        self,
+        *,
+        connection_string: str = None,
+        host: str = None,
+        port: int = None,
+        dbname: str = None,
+        connect_timeout: int = 10,
+        embedding_function: Callable = None,
+        metadata: dict = None,
+    ) -> None:
+        """
+        Initialize the vector database.
+
+        Note: connection_string or host + port + dbname must be specified
+
+        Args:
+            connection_string: "postgresql://username:password@hostname:port/database" | The PGVector connection string. Default is None.
+            host: str | The host to connect to. Default is None.
+            port: int | The port to connect to. Default is None.
+            dbname: str | The database name to connect to. Default is None.
+            connect_timeout: int | The timeout to set for the connection. Default is 10.
+            embedding_function: Callable | The embedding function used to generate the vector representation
+                of the documents. Default is None.
+            metadata: dict | The metadata of the vector database. Default is None. If None, it will use this
+                setting: {"hnsw:space": "ip", "hnsw:construction_ef": 30, "hnsw:M": 32}. For more details of
+                the metadata, please refer to [distances](https://github.com/nmslib/hnswlib#supported-distances),
+                [hnsw](https://github.com/chroma-core/chroma/blob/566bc80f6c8ee29f7d99b6322654f32183c368c4/chromadb/segment/impl/vector/local_hnsw.py#L184),
+                and [ALGO_PARAMS](https://github.com/nmslib/hnswlib/blob/master/ALGO_PARAMS.md).
```

Comment:
Is this true for pgvector?

Reply:
It is using the same metadata that is used in Chromadb implementation. I did not need to change anything there because the hashing function you implemented solved file versioning problem for me.

Reply:
The chromadb's metadata is for how to create index for the embedding search. It's different from the metadata of each document.

Reply:
Apologies, you're correct. I created an index for each of the supported types. eucledean, cosine, and inner product. I have updated the documentation.